src/axom/spin/tests/spin_bvh.cpp

```diff
@@ -1306,6 +1306,150 @@ void check_find_points_zip2d()
+ *        value to the checkMinDist lambda.
+ *
+ * \param [in] bvh The BVH object to use.
+ * \param [in] query_pts The array of query points to use against the BVH. 
+ */
+template <typename BVHType, typename PointType>
+void bvh_one_bbox_2d(BVHType& bvh,
+                     axom::Array<PointType>& points,
+                     axom::Array<PointType>& query_pts,
+                     bool none)
+{
+  EXPECT_TRUE(!query_pts.empty());
+
+  using FloatType = typename PointType::CoordType;
+  using BoxType = axom::primal::BoundingBox<FloatType, 2>;
+
+  // Borrowed from DistibutedClosestPoint.
+  struct MinCandidate
+  {
+    /// Squared distance to query point
+    double minSqDist {numerics::floating_point_limits<double>::max()};
+    /// Index within mesh of closest element
+    int minElem {-1};
+    /// MPI rank of closest element
+    int minRank {-1};
+  };
+
+  // Initialize the BVH with the points.
+  axom::IndexType npts = none ? 0 : points.size();
+  //axom::Array<BoxType> bboxes;
+  axom::Array<BoxType> bboxes;
+  for(axom::IndexType i = 0; i < npts; i++)
+    bboxes.push_back(BoxType(points[i]));
+  bvh.initialize(bboxes, bboxes.size());
+
+  // Call the BVH like the DistributedClosestPoint does.
+  auto it = bvh.getTraverser();
+
+  // Make a results array to contain the data values we read.
+  axom::Array<int> results(query_pts.size());
+  axom::ArrayView<int> results_view(results.data(), results.size());
+
+  // Loop over the query points.
+  axom::ArrayView<PointType> pointsView(points.data(), points.size());
+  axom::ArrayView<PointType> query_pts_view(query_pts.data(), query_pts.size());
+  npts = query_pts.size();
+  axom::for_all<typename BVHType::ExecSpaceType>(
+    npts,
+    AXOM_LAMBDA(int32 idx) mutable {
+      // Get the current query point.
+      auto qpt = query_pts_view[idx];
+      MinCandidate curr_min;
+
+      auto checkMinDist = [&](int32 current_node, const int32* leaf_nodes) {
+        int candidate_idx = leaf_nodes[current_node];
+        const PointType candidate_pt = pointsView[candidate_idx];
+        const double sq_dist = squared_distance(qpt, candidate_pt);
+
+        if(sq_dist < curr_min.minSqDist)
+        {
+          curr_min.minSqDist = sq_dist;
+          curr_min.minElem = candidate_idx;
+          curr_min.minRank = 0;  // rank
+        }
+      };
+
+      // Borrowed from DistributedClosestPoint.
+      auto traversePredicate = [&](const PointType& p, const BoxType& bb) -> bool {
+        auto sqDist = squared_distance(p, bb);
+        return sqDist <= curr_min.minSqDist;
+      };
+
+      // Traverse the tree, searching for the point with minimum distance.
+      it.traverse_tree(qpt, checkMinDist, traversePredicate);
+
+      // Save the index of the minElem.
+      results_view[idx] = curr_min.minElem;
+    });
+
+  // Make sure all of the indices we found are the expected value.
+  int expected_idx = none ? -1 : 0;
```

Comment:
Should we expose `-1` as a named value in `BVH` rather than as a magic number?

Reply:
I made a symbol that is more descriptive.